org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/serializer/impl/ChangeSerializer.java

```diff
@@ -140,9 +159,18 @@ protected RecordingResourceUpdater createResourceUpdater(Resource resource) {
 	}
 
 	protected void endRecordChanges(IAcceptor<IEmfResourceChange> changeAcceptor) {
+		endRecordChanges(changeAcceptor, null);
+	}
+
+	protected void endRecordChanges(IAcceptor<IEmfResourceChange> changeAcceptor, SubMonitor monitor) {
 		if (updaters.isEmpty()) {
 			return;
 		}
+		if (monitor == null) {
```

Comment:
I think you should use a dedicated SubMonitor here. Whatever gets passed in, you should convert either that or use this.monitor. But convert that to a `progress` SubMonitor. The type for `monitor` could be IProgressMonitor, no need to be more specific there.

Reply:
@kthoms @szarnekow @sailingKieler we should think about this point again and maybe correct the type in the signature